drivers/ethernet/eth_stm32_hal.c

```diff
@@ -32,19 +33,67 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #error DTCM for DMA buffer is activated but zephyr,dtcm is not present in dts
 #endif
 
+#ifndef PHY_ADDR
+#define PHY_ADDR	((uint32_t)CONFIG_ETH_STM32_HAL_PHY_ADDRESS)
+#endif
+
+#ifndef PHY_BSR
+#define PHY_BSR		((uint16_t)0x0001U)
+#endif
+
+#ifndef PHY_LINKED_STATUS
+#define PHY_LINKED_STATUS	((uint16_t)0x0004U)
+#endif
+
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+
+#define GET_FIRST_DMA_TX_DESC(heth)	(heth->Init.TxDesc)
+#define IS_ETH_DMATXDESC_OWN(dma_tx_desc)	(dma_tx_desc->DESC3 & \
+							ETH_DMATXNDESCRF_OWN)
+
+#define ETH_RXBUFNB	ETH_RX_DESC_CNT
+#define ETH_TXBUFNB	ETH_TX_DESC_CNT
+
+#define ETH_MEDIA_INTERFACE_MII		HAL_ETH_MII_MODE
+#define ETH_MEDIA_INTERFACE_RMII	HAL_ETH_RMII_MODE
+
+#define LL_AHB1_GRP1_PERIPH_ETHMAC	LL_AHB1_GRP1_PERIPH_ETH1MAC
+#define LL_AHB1_GRP1_PERIPH_ETHMACTX	LL_AHB1_GRP1_PERIPH_ETH1TX
+#define LL_AHB1_GRP1_PERIPH_ETHMACRX	LL_AHB1_GRP1_PERIPH_ETH1RX
+#define LL_AHB1_GRP1_PERIPH_ETHMACPTP	LL_AHB1_GRP1_PERIPH_ETH1MAC
+
+#define ETH_DMA_TRANSMIT_TIMEOUT	20U
```

Comment:
This timeout should be documented, maybe make it configurable.

Reply:
KConfig?

Reply:
I will put some description on this value, but suggest not to use it in KConfig. The main reason is interrupt based improvement form @lochej. This timeout value will be then eliminated. Are you okay with it?
P.S.: The suggested solution is already tested and prepared. @lochej and I just waiting until it this PR will be merged.

Reply:
It's ok then, but still a small description would be good.

Reply:
Corresponding description were added.